nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/recurrent/outputs/LSTMLayerOutputs.java

```diff
@@ -2,178 +2,132 @@
      */
-    private SDVariable f;
+    private SDVariable h;
 
-    /**
-     * Output - output gate activations.
-     * Shape depends on data format (in layer config):<br>
-     * TNS -> [timeSteps, batchSize, numUnits]<br>
-     * NST -> [batchSize, numUnits, timeSteps]<br>
-     * NTS -> [batchSize, timeSteps, numUnits]<br>
-     */
-    private SDVariable o;
 
     /**
-     * Output - input gate activations.
-     * Shape depends on data format (in layer config):<br>
-     * TNS -> [timeSteps, batchSize, numUnits]<br>
-     * NST -> [batchSize, numUnits, timeSteps]<br>
-     * NTS -> [batchSize, timeSteps, numUnits]<br>
+     * cell state at last step cL:
+     * [bS, nOut]   when directionMode FWD or BWD
+     * 2, bS, nOut] when directionMode  BIDIR_SUM, BIDIR_CONCAT or BIDIR_EXTRA_DIM
      */
-    private SDVariable z;
+    private SDVariable cL;
 
     /**
-     * Cell state, post tanh.
-     * Shape depends on data format (in layer config):<br>
-     * TNS -> [timeSteps, batchSize, numUnits]<br>
-     * NST -> [batchSize, numUnits, timeSteps]<br>
-     * NTS -> [batchSize, timeSteps, numUnits]<br>
+     * output at last step hL:
+     * [bS, nOut]   when directionMode FWD or BWD
+     * 2, bS, nOut] when directionMode  BIDIR_SUM, BIDIR_CONCAT or BIDIR_EXTRA_DIM
      */
-    private SDVariable h;
+    private SDVariable hL;
 
-    /**
-     * Current cell output.
-     * Shape depends on data format (in layer config):<br>
-     * TNS -> [timeSteps, batchSize, numUnits]<br>
-     * NST -> [batchSize, numUnits, timeSteps]<br>
-     * NTS -> [batchSize, timeSteps, numUnits]<br>
-     */
-    private SDVariable y;
-
-    public LSTMLayerOutputs(SDVariable[] outputs, RnnDataFormat dataFormat){
-        Preconditions.checkArgument(outputs.length == 7,
-                "Must have 7 LSTM layer outputs, got %s", outputs.length);
-
-        i = outputs[0];
-        c = outputs[1];
-        f = outputs[2];
-        o = outputs[3];
-        z = outputs[4];
-        h = outputs[5];
-        y = outputs[6];
-        this.dataFormat = dataFormat;
+    private boolean hasFullSequence(LSTMLayerConfig lstmLayerConfig) {
+        return lstmLayerConfig.isRetFullSequence();
     }
 
-    /**
-     * Get all outputs returned by the cell.
-     */
-    public List<SDVariable> getAllOutputs(){
-        return Arrays.asList(i, c, f, o, z, h, y);
+    private boolean hasLastC(LSTMLayerConfig lstmLayerConfig) {
+        return lstmLayerConfig.isRetLastC();
     }
 
-    /**
-     * Get y, the output of the cell for all time steps.
-     * 
-     * Shape depends on data format (in layer config):<br>
-     * TNS -> [timeSteps, batchSize, numUnits]<br>
-     * NST -> [batchSize, numUnits, timeSteps]<br>
-     * NTS -> [batchSize, timeSteps, numUnits]<br>
-     */
-    public SDVariable getOutput(){
-        return y;
+    private boolean hasLastH(LSTMLayerConfig lstmLayerConfig) {
+        return lstmLayerConfig.isRetLastH();
     }
```

Comment:
As noted below - delete hasFullSequence, hasLastC, and hasLastH methods - they are unnecessary.